assembl/graphql/utils.py

```diff
@@ -189,3 +192,46 @@ def update_attachment(discussion, attachment_model, new_value, attachments, atta
             attachmentPurpose=attachment_purpose
         )
         attachments.append(attachment)
+
+
+def get_posts_for_phases(discussion, identifiers):
+    """Return related posts for the given phases `identifiers` on `discussion`.
+    """
+    # Retrieve the phases with posts
+    identifiers_with_posts = [i for i in identifiers if i in PHASES_WITH_POSTS]
+    if not discussion or not identifiers_with_posts:
+        return None
+
+    ideas = []
+    # If survey phase, we need the root thematic
+    if Phases.survey.value in identifiers_with_posts:
+        root_thematic = get_root_thematic_for_phase(discussion, Phases.survey.value)
+        if root_thematic:
+            ideas.append(root_thematic)
+
+        identifiers_with_posts.remove(Phases.survey.value)
+
+    if identifiers_with_posts:
+        is_multi_columns = Phases.multiColumns.value in identifiers_with_posts and \
+            len(identifiers_with_posts) == 1
+        ideas.extend(
+            get_ideas(
+                discussion.id,
+                Phases.multiColumns.value if is_multi_columns else None
+            ).all()
+        )
+
+    if not ideas:
+        return None
+
+    model = models.AssemblPost
+    query = discussion.db.query(model)
+    query_source = query
+    for index, idea in enumerate(ideas):
+        related = idea.get_related_posts_query(True)
+        related_query = query_source.join(
+            related, model.id == related.c.post_id
+        )
```

Comment:
You currently include deleted posts here. You may want to add an option to filter on `publication_state == models.PublicationStates.PUBLISHED` (see `get_published_posts`)

Reply:
Please see get_related_posts_query => include_deleted=False

Reply:
include_deleted is False by default, right, but it's actually not sufficient. This include_deleted=False means adding the filter `query.filter(content.tombstone_date == None)` but we don't tombstone delete posts having a parent anymore, so you need an additional filter on publication_state.

Reply:
done